python/brainyflow.py

```
@@ -1,65 +1,348 @@
+            return self.node.on(self.action, other)
+
+    def get_next_nodes(self, action=DEFAULT_ACTION):
+        """Get successor nodes for a specific action."""
+        next_nodes = self.successors.get(action, [])
+        if not next_nodes and action != DEFAULT_ACTION and self.successors:
+            warnings.warn(f"Flow ends: '{action}' not found in {list(self.successors.keys())}")
+        return next_nodes
+    
+    async def prep(self, memory):
+        """Prepare phase - override in subclasses."""
+        pass
```

Ruff (0.8.2)

`BaseNode.prep` is an empty method in an abstract base class, but has no abstract decorator

(B027)